runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -418,6 +418,18 @@ public void stop() {
         }
     }
 
+    /**
+     * Reestablish the netty connection.
+     */
+    @Override
+    public void reconnect() {
+        // Close the channel and auto-trigger the reconnection callback.
```

Comment:
I think the interaction between the thread calling reconnect and the channel worker thread can cause some concurrency issues where there could be threads waiting on `connectionFuture` and then `addReconnectionOnCloseFuture` would swap the reference without completing it exceptionally. 

It's not clear that this is a practical issue because external threads would timeout (because of `connectionFuture.get(parameters.getConnectionTimeout().toMillis(), TimeUnit.MILLISECONDS);`)